classes/data-stores/ActionScheduler_DBStore.php

```diff
@@ -268,78 +268,113 @@ protected function get_query_actions_sql( array $query, $select_or_count = 'sele
 			throw new InvalidArgumentException( __( 'Invalid value for select or count parameter. Cannot query actions.', 'action-scheduler' ) );
 		}
 
-		$query = wp_parse_args( $query, [
-			'hook'             => '',
-			'args'             => null,
-			'date'             => null,
-			'date_compare'     => '<=',
-			'modified'         => null,
-			'modified_compare' => '<=',
-			'group'            => '',
-			'status'           => '',
-			'claimed'          => null,
-			'per_page'         => 5,
-			'offset'           => 0,
-			'orderby'          => 'date',
-			'order'            => 'ASC',
-		] );
+		$query = wp_parse_args( $query, array(
+			'hook'             		=> '',
+			'args'             		=> null,
+			'args_partial_match'	=> false,
+			'date'             		=> null,
+			'date_compare'     		=> '<=',
+			'modified'         		=> null,
+			'modified_compare' 		=> '<=',
+			'group'            		=> '',
+			'status'           		=> '',
+			'claimed'          		=> null,
+			'per_page'         		=> 5,
+			'offset'           		=> 0,
+			'orderby'          		=> 'date',
+			'order'            		=> 'ASC',
+		 ) );
 
 		/** @var \wpdb $wpdb */
 		global $wpdb;
+		$mysql_version = $wpdb->db_version();
 		$sql  = ( 'count' === $select_or_count ) ? 'SELECT count(a.action_id)' : 'SELECT a.action_id';
 		$sql .= " FROM {$wpdb->actionscheduler_actions} a";
-		$sql_params = [];
+		$sql_params = array();
 
-		if ( ! empty( $query[ 'group' ] ) || 'group' === $query[ 'orderby' ] ) {
+		if ( ! empty( $query['group'] ) || 'group' === $query['orderby'] ) {
 			$sql .= " LEFT JOIN {$wpdb->actionscheduler_groups} g ON g.group_id=a.group_id";
 		}
 
 		$sql .= " WHERE 1=1";
 
-		if ( ! empty( $query[ 'group' ] ) ) {
+		if ( ! empty( $query['group'] ) ) {
 			$sql          .= " AND g.slug=%s";
-			$sql_params[] = $query[ 'group' ];
+			$sql_params[] = $query['group'];
 		}
 
-		if ( $query[ 'hook' ] ) {
+		if ( $query['hook'] ) {
 			$sql          .= " AND a.hook=%s";
-			$sql_params[] = $query[ 'hook' ];
-		}
-		if ( ! is_null( $query[ 'args' ] ) ) {
-			$sql          .= " AND a.args=%s";
-			$sql_params[] = $this->get_args_for_query( $query[ 'args' ] );
+			$sql_params[] = $query['hook'];
+		}
+
+		if ( ! is_null( $query['args'] ) ) {
+			if ( isset( $query['args_partial_match'] ) && true === $query['args_partial_match'] && $mysql_version >= '5.7' ) {
+				foreach ( $query['args'] as $key => $value ) {
+					$supported_types = array( 'integer', 'boolean', 'double', 'string' );
+					if ( ! in_array( gettype( $value ), $supported_types ) ) {
+						continue;
+					}
+					switch ( gettype( $value ) ) {
```

Comment:
For simplicity, instead of this `switch` with 4 almost identical cases, we could add the placeholders directly to the `$supported_types` array (such as `'double' => '%f'`) and use that in the `$sql` var.